src/main/java/uk/gov/ida/verifyserviceprovider/validators/SubjectValidator.java

```diff
@@ -52,6 +54,15 @@ public void validate(Subject subject, String expectedInResponseTo) {
             throw new SamlResponseValidationException(String.format("'InResponseTo' must match requestId. Expected %s but was %s", expectedInResponseTo, actualInResponseTo));
         }
 
+        String recipient = subjectConfirmationData.getRecipient();
+        if (recipient == null) {
+            throw new SamlResponseValidationException("Subject confirmation data must contain 'Recipient'.");
+        }
+
+        if (!assertionConsumerServiceUri.equals(recipient)) {
+            throw new SamlResponseValidationException(String.format("'Recipient' must match entity id. Expected %s but was %s", assertionConsumerServiceUri, recipient));
```

Comment:
Error message should say "'Recipient' must match assertionConsumerServiceUri" not "entity id" right?

Reply:
Done